### PR TITLE
Feat: Auth, private route,

### PR DESCRIPTION
**Pull request by pritam-kr**
- Create login, signup authentication 
- store uid in local storage
- created a private route 
- [profile page](https://github.com/pritam-kr/quizzie/pull/2/commits/9f6fda5e780b919e7a8c10350e910d10b67e130b)

### CODE REVIEW

**Review by lannisterrr — COMMENTED** [review]
Good structured code.
Like the use of type checking .
overall good to merge.

File: src/allTypes/authTypes.ts [hunk 1]
```diff
@@ -1,5 +1,40 @@
...
+  isAuthenticated: boolean;
+};
+
+// export type AuthenticationAction = {
+//   type: "SET_AUTHENTICATION";
+//   payload: {
+//     isAuthenticated?: boolean;
+//     uid: string;
+//   };
+// };
```
Comment (msdeshmukh009) [inline src/allTypes/authTypes.ts:19]: Remove the comment if not necessary.

**Review by msdeshmukh009 — APPROVED** [review]
The code looks good. Think of adding a live preview.

**Review by varnita619 — APPROVED** [review]
Good folder structure with clean code
Ready to merge

**Pull request merged**